imbi/automations.py

```diff
@@ -173,3 +188,99 @@ async def _get_gitlab_parent(self, project: Project) -> dict:
                             project.namespace.gitlab_group_name,
                             project.project_type.gitlab_project_prefix)
         return gitlab_parent
+
+
+class GitLabInitialCommitAutomation(Automation):
+    def __init__(self,
+                 automation_settings: dict,
+                 project_id: int,
+                 cookie_cutter: str,
+                 user: 'imbi.user.User',
+                 db: sprockets_postgres.PostgresConnector):
+        super().__init__(user, db)
+        self._cookie_cutter_name = cookie_cutter
+        self._imbi_project_id = project_id
+
+        self._cookie_cutter: typing.Union[CookieCutter, None] = None
+        self._gitlab_project_info: typing.Union[dict, None] = None
+        self._project: typing.Union[Project, None] = None
+        self._token: typing.Union[imbi.integrations.IntegrationToken,
+                                  None] = None
+
+    async def prepare(self) -> typing.List[str]:
+        self._project = await self._get_project(self._imbi_project_id)
+        self._token = await self._get_gitlab_token()
+        self._cookie_cutter = await self._get_cookie_cutter(
+            self._cookie_cutter_name)
+
+        if not self._has_error() and (self._project.project_type.id !=
```

Comment:
Why do we check `self._has_error()` here? In what situation would this error sometimes be handled already and other times need to be handled here?